cedar-policy/src/api.rs

```diff
@@ -1812,6 +1837,14 @@ impl PolicySet {
         Ok(())
     }
 
+    /// Unlink a dynamic policy from the policy set
+    pub fn unlink(&mut self, policy_id: PolicyId) -> Result<(), PolicySetError> {
+        match self.policies.remove(&policy_id) {
+            Some(_) => Ok(()),
+            None => Err(PolicySetError::UnlinkingError(policy_id)),
+        }
```

Comment:
same here, I think we need to also remove from `self.ast` and add testing that authz queries are affected